compiler/rustc_middle/src/ty/consts/kind.rs

```diff
@@ -68,6 +72,63 @@ pub enum ConstKind<'tcx> {
     /// A placeholder for a const which could not be computed; this is
     /// propagated to avoid useless error messages.
     Error(ty::DelaySpanBugEmitted),
+
+    /// Abstract which contains an expression which has partially evaluated items.
+    Abstract(Expr<'tcx>),
```

Comment:
From the looks of it, "symbolic" would be a better name. 'Param' is already an "abstract" const.

Reply:
Also, in which sense is this "empty"? The PR description talks about "empty".

Reply:
Initially I was only go to do the plumbing w/o an actual implementation, but decided to do an actual impl

Reply:
`Symbolic` doesn't seem to bad, would be fine  with that. I agree that `Abstract` isn't a great name here 🤔 

Alternatively `ConstKind::Expr` maybe